### ISSUE TITLE
Two issues with overring functions in the global namespace

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#330]
Originally reported by: **Antony Lee (BitBucket: [anntzer](http://bitbucket.org/anntzer), GitHub: @anntzer?)**

---

```
#!python
func1 = None
def func2(): pass

def override_func():
    global func1, func2
    def func1(): pass
    func2 = None
```

Pylint gives various spurious warnings here in `override_func`: it does not realize that `def func1():...` assigns a new value at the global level too, and thinks that there is no previous binding for `func2` before overriding it.

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/330
